src/task.ts

```diff
@@ -36,53 +38,78 @@ export function isTask(candidate: any): candidate is Task<any> {
     typeof candidate.listen === 'function' &&
     typeof candidate.cancel === 'function' &&
     typeof candidate.isRunning === 'boolean' &&
+    typeof candidate.isFinished === 'boolean' &&
     // todo test if that works
     // typeof candidate.isIdle === 'boolean' &&
     typeof candidate.isCancelled === 'boolean'
   );
 }
 
-export const IdleTask: Task<any> = {
-  isRunning: false,
-  isCancelled: false,
-  isIdle: true,
-  cancel() {},
-  listen() {},
-  toPromise() {
-    return Promise.resolve();
-  },
-};
+export function idleTask(): Task<any> {
+  return task(() => {});
+}
 
 export function task<TResult, TReason = any>(
   runner: TaskRunner<TResult, TReason>
 ): Task<TResult, TReason> {
-  const subscribers = new Set<TaskEventNotifications<TResult, TReason>>();
-  const task: Task<TResult, TReason> = {
-    isRunning: true,
+  const eventSubscribers: TaskEventNotifications<TResult, TReason>[] = [];
+  const cancelCallbacks: { (): void }[] = [];
```

Comment:
(Nit): Just personal preference. I prefer to use `Array<...>` to annotate arrays as I feel like the `[]` gets lost too easily in all of the type annotations. Especially when generics are involved.

```suggestion
  const eventSubscribers: Array<TaskEventNotifications<TResult, TReason>> = [];
  const cancelCallbacks: Array<() => void> = [];
```